### ISSUE TITLE
Developer builds don't run on Windows due to missing libwinpthread-1.dll

### ISSUE THREAD

**Issue by thoni56** [clementine-player/Clementine#6270]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [x] I checked the issue tracker for similar issues
- [x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved _(Duh! Changelog only lists up to 1.3...)_
- [x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/) _(That's where I found the problem since there are no other recent releases...)_

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: Windows 10
- Clementine version: ClementineSetup-1.3.1-552-gc4d6424

### Expected behaviour / actual behaviour

![bild](https://user-images.githubusercontent.com/204496/51686105-06e82500-1ff0-11e9-8e91-94049ebdee6a.png)

### Steps to reproduce the problem (only for bugs)

Just run Clementine Windows developer build.

**Comment by chucku** [comment]
I have found the same problem...

**Comment by mkserge** [comment]
Same problem here.

**Comment by GATOQSECOMIO** [comment]
same problem here

**Comment by Venryx** [comment]
Same problem here. This is an important issue because it means the latest Windows builds are unusable, forcing us to go all the way back to the 2016 official release! (which doesn't support Opus files 😞)

**Comment by mkserge** [comment]
> Same problem here. This is an important issue because it means the latest Windows builds are unusable, forcing us to go all the way back to the 2016 official release! (which doesn't support Opus files 😞)

Not ideal but you can manually copy-paste the missing lib from the working installation. Basically install an earlier version of Clementine, copy libwinthread, uninstall then install the latest version of Clementine and finally copy the library back into the main directory.

**Comment by hatstand** [comment]
If someone actually wants to work towards fixing this, a good start would be moving more builds onto travis instead of our self-hosted, flaky build infrastructure.

**Comment by chucku** [comment]
1.3.1-388-gf1b767f is last version I got to work...

Regards,
Chuck

**Comment by Unknown-Zombie** [comment]
The developer builds work fine on my computer, and they have done so for a long time. I found out that I'm able to run them because I have TDM-GCC-64 installed on my computer.

This should be addressed in a proper manner, but a workaround until then for layman users would be to install [TDM-GCC](http://tdm-gcc.tdragon.net/download).

**Comment by JulianVolodia** [comment]
> If someone actually wants to work towards fixing this, a good start would be moving more builds onto travis instead of our self-hosted, flaky build infrastructure.

@hatstand I will do this integration in following week.

**Comment by JulianVolodia** [comment]
@hatstand as I don't saw any issue/thread I will create new one before sending PR. If I missed that issue, please tell me about it.

**Comment by JulianVolodia** [comment]
Dear @hatstand 
I have a questions to you as you are guru to me. 

Will we drop version with qt4 at all?

Shall I work on merge (integrate CI and merge changes) from branch qt5-mac to qt5?

I will need feedback about that integration from you. So, are you more fluent with Circle, Travis if need tips? Or whatever?
You said about CircleCI, but first I will try to make working build on TravisCI because of:
- we will have faster building (because synergy of 2 free for open source project platform credits and parallel build workers)
- I found some working example of integration Qt5 build on TravisCI

**Comment by hatstand** [comment]
`qt5-mac` is merged into `qt5` already IIRC.

We're already successfully doing Linux & Mac builds for qt5 on CircleCI so it's just windows left. You're welcome to do it on travis if you like and I can port it afterwards but the windows build is _hard_.

I think it's likely that I'll just merge the qt5 branch to master soon and then we can figure out reliable builds just for qt5.

**Comment by JulianVolodia** [comment]
> `qt5-mac` is merged into `qt5` already IIRC.

Yes, that was my one big mistake cause qt5-mac is in qt5. Sorry. 

> windows build is _hard_.

I remember. I done it one time. Have some ice breaking tips for Windows? On which version of MS product you build it? I will try do as similar to your workflow.

**Comment by JulianVolodia** [comment]
And last but not least, which other old builder platforms you want to move to new one?

**Comment by hatstand** [comment]
I started hacking away at it here: https://github.com/clementine-player/Clementine/compare/circleci-mingw

Every other platform is sorted on circleci I think including mac, ubuntu, debian and fedora.

**Comment by JulianVolodia** [comment]
BTW. I checked Clementine-qt5 AUR package also (I use Arch Linux, Debian and Windows as workstations).

What you think about checking against Arch Linux also. There are some ppl out there using it, and also could be interesing step for their community to support us. They probably more likely to provide faster and good feedback because of more advanced distro used.

**Comment by hatstand** [comment]
I don't know much about Arch but you're welcome to setup builds for it.

**Comment by JulianVolodia** [comment]
> I started hacking away at it here: https://github.com/clementine-player/Clementine/compare/circleci-mingw

Thanks for info about that. 👍 

> I don't know much about Arch but you're welcome to setup builds for it.

TL;DR there is 2 repositories, one similar to Debian's - then using pacman as apt. And second, AUR, from where you acquire package with some metadata (PKGBUILD file included) and run makepkg. AUR package clementine-qt5 is depend on qt5 branch so if something will break they will have fast feedback about it.

**Comment by JulianVolodia** [comment]
> Unfortunately, once I had copied this into the Clementine install directory, it then failed to start with error 0xc000007b

https://social.msdn.microsoft.com/Forums/vstudio/en-US/fba7d3c0-b595-42be-8e17-7fd0468565ca/the-application-was-unable-to-start-correctly-0xc000007b-click-ok-to-close-the-application?forum=vsdebug

@jjberry314 have you tried to do sth with it? Are you programmer? Did you overcome problem?

**Comment by JulianVolodia** [comment]
> 1.3.1-388-gf1b767f is last version I got to work...

@chucku could you tell more about system version and your configuration?

**Comment by JulianVolodia** [comment]
Hi @hatstand so next step is to refactor Windows version build to support Qt5?

**Comment by JulianVolodia** [comment]
@jjberry314 I'm sorry, I hope you will be happy with that. Share the best music you like there or on my email. I looking for new tastes.

**Comment by JulianVolodia** [comment]
> I don't know much about Arch but you're welcome to setup builds for it.

I didn't know that @Chocobozzz is the Maintainer of clementine-qt5-git @ AUR ;D
@Chocobozzz , what you think about this idea of CircleCI builds for Arch Linux against master? Probably clementine-git should be overtaken by clementine-qt5-git now.

**Comment by hatstand** [comment]
https://builds.clementine-player.org/windows/

**Comment by chucku** [comment]
That seems to work fine!  Thank you very much :)
